src/globals.c

```diff
@@ -198,6 +194,10 @@ bool set_format_string(char *userinput)
 {
 	int ret;
 
+	if (format_string) {
+		return true;
+	}
```

Comment:
I did miss that, but it looks like the check on 220 only catches the NULL case. Given the new approach, which allows for multiple calls with a non-NULL arg, I need to move that check up to cover userinput being non-NULL as well.